include/networkit/graph/Graph.hpp

```diff
@@ -445,11 +445,10 @@ class Graph final {
     /**
      * Class to iterate over the in/out neighbors of a node.
      */
-    class NeighborIterator {
+    class NeighborIterator
+        : public std::iterator<std::forward_iterator_tag, const node> {
```

Comment:
`std::iterator` is deprecated in C++17. It does nothing except for defining the member types listed in <https://en.cppreference.com/w/cpp/iterator/iterator>, so it should be easy to avoid.